matrix/inverse_of_matrix.py

```diff
@@ -25,24 +26,137 @@ def inverse_of_matrix(matrix: list[list[float]]) -> list[list[float]]:
+        swapped_matrix[1][0], swapped_matrix[0][1] = -matrix[1][0], -matrix[0][1]
+
+        # Calculate the inverse of the matrix
+        return [
+            [float(d(n) / determinant) or 0.0 for n in row] for row in swapped_matrix
+        ]
+    elif (
+        len(matrix) == 3
+        and len(matrix[0]) == 3
+        and len(matrix[1]) == 3
+        and len(matrix[2]) == 3
+    ):
+        # Calculate the determinant of the matrix using Sarrus rule
+        determinant = (
+            (d(matrix[0][0]) * d(matrix[1][1]) * d(matrix[2][2]))
+            + (d(matrix[0][1]) * d(matrix[1][2]) * d(matrix[2][0]))
+            + (d(matrix[0][2]) * d(matrix[1][0]) * d(matrix[2][1]))
+        ) - (
+            (d(matrix[0][2]) * d(matrix[1][1]) * d(matrix[2][0]))
+            + (d(matrix[0][1]) * d(matrix[1][0]) * d(matrix[2][2]))
+            + (d(matrix[0][0]) * d(matrix[1][2]) * d(matrix[2][1]))
+        )
+        if determinant == 0:
+            raise ValueError("This matrix has no inverse.")
+
+        # Creating cofactor matrix
+        cofactor_matrix = [
+            [d(0.0), d(0.0), d(0.0)],
+            [d(0.0), d(0.0), d(0.0)],
+            [d(0.0), d(0.0), d(0.0)],
+        ]
+        cofactor_matrix[0][0] = (d(matrix[1][1]) * d(matrix[2][2])) - (
+            d(matrix[1][2]) * d(matrix[2][1])
+        )
+        cofactor_matrix[0][1] = -(
+            (d(matrix[1][0]) * d(matrix[2][2])) - (d(matrix[1][2]) * d(matrix[2][0]))
+        )
+        cofactor_matrix[0][2] = (d(matrix[1][0]) * d(matrix[2][1])) - (
+            d(matrix[1][1]) * d(matrix[2][0])
+        )
+        cofactor_matrix[1][0] = -(
+            (d(matrix[0][1]) * d(matrix[2][2])) - (d(matrix[0][2]) * d(matrix[2][1]))
+        )
+        cofactor_matrix[1][1] = (d(matrix[0][0]) * d(matrix[2][2])) - (
+            d(matrix[0][2]) * d(matrix[2][0])
+        )
+        cofactor_matrix[1][2] = -(
+            (d(matrix[0][0]) * d(matrix[2][1])) - (d(matrix[0][1]) * d(matrix[2][0]))
+        )
+        cofactor_matrix[2][0] = (d(matrix[0][1]) * d(matrix[1][2])) - (
+            d(matrix[0][2]) * d(matrix[1][1])
+        )
+        cofactor_matrix[2][1] = -(
+            (d(matrix[0][0]) * d(matrix[1][2])) - (d(matrix[0][2]) * d(matrix[1][0]))
+        )
+        cofactor_matrix[2][2] = (d(matrix[0][0]) * d(matrix[1][1])) - (
+            d(matrix[0][1]) * d(matrix[1][0])
+        )
 
-    # Calculate the determinant of the matrix
-    determinant = d(matrix[0][0]) * d(matrix[1][1]) - d(matrix[1][0]) * d(matrix[0][1])
-    if determinant == 0:
-        raise ValueError("This matrix has no inverse.")
+        # Transpose the cofactor matrix (Adjoint matrix)
+        adjoint_matrix = [
+            [d(0.0), d(0.0), d(0.0)],
+            [d(0.0), d(0.0), d(0.0)],
+            [d(0.0), d(0.0), d(0.0)],
+        ]
+        for i in range(3):
+            for j in range(3):
+                adjoint_matrix[i][j] = cofactor_matrix[j][i]

```

Comment:
```suggestion
        adjoint_matrix = zip(*cofactor_matrix)
```
https://www.geeksforgeeks.org/transpose-matrix-single-line-python

Reply:
Ok sir.